src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        String randomColor = ColorSupplier.getRandomColor();
+        // if getRandomColor() not be static, will be next problem:
+        // Non-static method 'getRandomColor()' cannot be referenced from a static context
+        final String[] figures = {"square",
+                                  "rectangle",
+                                  "right triangle",
+                                  "circle",
+                                  "isosceles trapezoid"};
+        Random random = new Random();
+        String randomFigureType = figures[random.nextInt(figures.length)];
+
+        switch (randomFigureType) {
+            case "square":
+                double side = Math.random() * 10;
```

Comment:
think about extracting logic about creating a random square into a private method.

Reply:
what do you mean?

Reply:
lets create a separate private method for each type of figure, simply call it here,
this would make your code easier to read

```getRandomSquare()```